lib/features/labels/view/pages/labels_page.dart

```diff
@@ -257,98 +259,70 @@ class _LabelsPageState extends State<LabelsPage> with SingleTickerProviderStateM
     Navigator.push(
       context,
-      MaterialPageRoute(
-        builder: (_) => RepositoryProvider.value(
-          value: context.read<LabelRepository>(),
-          child: EditCorrespondentPage(correspondent: correspondent),
-        ),
-      ),
+      buildLabelPageRoute(EditCorrespondentPage(correspondent: correspondent)),
     );
   }
 
   void _openEditDocumentTypePage(DocumentType docType) {
     Navigator.push(
       context,
-      MaterialPageRoute(
-        builder: (_) => RepositoryProvider.value(
-          value: context.read<LabelRepository>(),
-          child: EditDocumentTypePage(documentType: docType),
-        ),
-      ),
+      buildLabelPageRoute(EditDocumentTypePage(documentType: docType)),
     );
   }
 
   void _openEditTagPage(Tag tag) {
     Navigator.push(
       context,
-      MaterialPageRoute(
-        builder: (_) => RepositoryProvider.value(
-          value: context.read<LabelRepository>(),
-          child: EditTagPage(tag: tag),
-        ),
-      ),
+      buildLabelPageRoute(EditTagPage(tag: tag)),
     );
   }
 
   void _openEditStoragePathPage(StoragePath path) {
     Navigator.push(
       context,
-      MaterialPageRoute(
-        builder: (_) => RepositoryProvider.value(
-          value: context.read<LabelRepository>(),
-          child: EditStoragePathPage(
-            storagePath: path,
-          ),
-        ),
-      ),
+      buildLabelPageRoute(EditStoragePathPage(
+        storagePath: path,
+      )),
     );
   }
 
   void _openAddCorrespondentPage() {
     Navigator.push(
       context,
-      MaterialPageRoute(
-        builder: (_) => RepositoryProvider.value(
-          value: context.read<LabelRepository>(),
-          child: const AddCorrespondentPage(),
-        ),
-      ),
+      buildLabelPageRoute(const AddCorrespondentPage()),
     );
   }
 
   void _openAddDocumentTypePage() {
     Navigator.push(
       context,
-      MaterialPageRoute(
-        builder: (_) => RepositoryProvider.value(
-          value: context.read<LabelRepository>(),
-          child: const AddDocumentTypePage(),
-        ),
-      ),
+      buildLabelPageRoute(const AddDocumentTypePage()),
     );
   }
 
   void _openAddTagPage() {
     Navigator.push(
       context,
-      MaterialPageRoute(
-        builder: (_) => RepositoryProvider.value(
-          value: context.read<LabelRepository>(),
-          child: const AddTagPage(),
-        ),
-      ),
+      buildLabelPageRoute(const AddTagPage()),
     );
   }
 
   void _openAddStoragePathPage() {
     Navigator.push(
       context,
-      MaterialPageRoute(
-        builder: (_) => RepositoryProvider.value(
-          value: context.read<LabelRepository>(),
-          child: const AddStoragePathPage(),
-        ),
-      ),
+      buildLabelPageRoute(const AddStoragePathPage()),
+    );
+  }
+
+  MaterialPageRoute<dynamic> buildLabelPageRoute(Widget page) {
+    return MaterialPageRoute(
+        builder: (_) => MultiBlocProvider(
```

Comment:
Please change this to MultuProvider as no Blocs are provided. Just a matter of semantics ;)